From 1600ddca50579c876163f7bdf23e62a7c6c9e0c6 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:07:26 -0400
Subject: [usb] cdc-acm: Remove Motorola/Telit H24 serial interfaces from ACM driver

Message-id: <1410887463-20674-375-git-send-email-dzickus@redhat.com>
Patchwork-id: 95444
O-Subject: [RHEL7 PATCH 374/591] USB: cdc-acm: Remove Motorola/Telit H24 serial interfaces from ACM driver
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 895d240d1db0b2736d779200788e4c4aea28a0c6
Author: Michael Ulbricht <michael.ulbricht@systec-electronic.com>
Date:   Tue Mar 25 10:34:18 2014 +0100

    USB: cdc-acm: Remove Motorola/Telit H24 serial interfaces from ACM driver

    By specifying NO_UNION_NORMAL the ACM driver does only use the first two
    USB interfaces (modem data & control). The AT Port, Diagnostic and NMEA
    interfaces are left to the USB serial driver.

    Signed-off-by: Michael Ulbricht <michael.ulbricht@systec-electronic.com>
    Signed-off-by: Alexander Stein <alexander.stein@systec-electronic.com>
    Signed-off-by: Oliver Neukum <oliver@neukum.org>
    Cc: stable <stable@vger.kernel.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/class/cdc-acm.c b/drivers/usb/class/cdc-acm.c
index d5d2c92..904efb6 100644
--- a/drivers/usb/class/cdc-acm.c
+++ b/drivers/usb/class/cdc-acm.c
@@ -1654,13 +1654,27 @@ static const struct usb_device_id acm_ids[] = {
  },
  /* Motorola H24 HSPA module: */
  { USB_DEVICE(0x22b8, 0x2d91) }, /* modem                                */
- { USB_DEVICE(0x22b8, 0x2d92) }, /* modem           + diagnostics        */
- { USB_DEVICE(0x22b8, 0x2d93) }, /* modem + AT port                      */
- { USB_DEVICE(0x22b8, 0x2d95) }, /* modem + AT port + diagnostics        */
- { USB_DEVICE(0x22b8, 0x2d96) }, /* modem                         + NMEA */
- { USB_DEVICE(0x22b8, 0x2d97) }, /* modem           + diagnostics + NMEA */
- { USB_DEVICE(0x22b8, 0x2d99) }, /* modem + AT port               + NMEA */
- { USB_DEVICE(0x22b8, 0x2d9a) }, /* modem + AT port + diagnostics + NMEA */
+ { USB_DEVICE(0x22b8, 0x2d92),   /* modem           + diagnostics        */
+ .driver_info = NO_UNION_NORMAL, /* handle only modem interface          */
+ },
+ { USB_DEVICE(0x22b8, 0x2d93),   /* modem + AT port                      */
+ .driver_info = NO_UNION_NORMAL, /* handle only modem interface          */
+ },
+ { USB_DEVICE(0x22b8, 0x2d95),   /* modem + AT port + diagnostics        */
+ .driver_info = NO_UNION_NORMAL, /* handle only modem interface          */
+ },
+ { USB_DEVICE(0x22b8, 0x2d96),   /* modem                         + NMEA */
+ .driver_info = NO_UNION_NORMAL, /* handle only modem interface          */
+ },
+ { USB_DEVICE(0x22b8, 0x2d97),   /* modem           + diagnostics + NMEA */
+ .driver_info = NO_UNION_NORMAL, /* handle only modem interface          */
+ },
+ { USB_DEVICE(0x22b8, 0x2d99),   /* modem + AT port               + NMEA */
+ .driver_info = NO_UNION_NORMAL, /* handle only modem interface          */
+ },
+ { USB_DEVICE(0x22b8, 0x2d9a),   /* modem + AT port + diagnostics + NMEA */
+ .driver_info = NO_UNION_NORMAL, /* handle only modem interface          */
+ },
 
  { USB_DEVICE(0x0572, 0x1329), /* Hummingbird huc56s (Conexant) */
  .driver_info = NO_UNION_NORMAL, /* union descriptor misplaced on
-- 
1.7.1